File: app/src/main/java/com/elishaazaria/sayboard/ime/recognizers/providers/VoskServerProvider.java
package com.elishaazaria.sayboard.ime.recognizers.providers;

import com.elishaazaria.sayboard.data.VoskServerData;
import com.elishaazaria.sayboard.ime.recognizers.RecognizerSource;
import com.elishaazaria.sayboard.ime.recognizers.VoskServer;
import com.elishaazaria.sayboard.preferences.ModelPreferences;

import java.util.List;

public class VoskServerProvider implements RecognizerSourceProvider{
    @Override
    public void loadSources(List<RecognizerSource> recognizerSources) {
        for (VoskServerData voskServer : ModelPreferences.getVoskServers()) {
            recognizerSources.add(new VoskServer(voskServer));
        }
    }
}

File: app/src/main/java/com/elishaazaria/sayboard/ime/recognizers/providers/VoskServerProvider.kt
package com.elishaazaria.sayboard.ime.recognizers.providers

import com.elishaazaria.sayboard.data.InstalledModelReference
import com.elishaazaria.sayboard.ime.recognizers.RecognizerSource

class VoskServerProvider : RecognizerSourceProvider {
    override fun getInstalledModels(): List<InstalledModelReference> {
        TODO("Not yet implemented")
    }

    override fun recognizerSourceForModel(localModel: InstalledModelReference): RecognizerSource? {
        TODO("Not yet implemented")
    }
}